lib/ansible/plugins/connection/__init__.py

```diff
@@ -285,6 +285,24 @@ def update_vars(self, variables: dict[str, t.Any]) -> None:
                 display.debug('Set connection var {0} to {1}'.format(varname, value))
                 variables[varname] = value
 
+    # TODO find better name and place for this. Public API? On AnsiblePlugin?
+    def _var_options(self, variables, templar):
```

Comment:
IMO it shouldn't be PublicAPI, and it does makes more sense here than AnsiblePlugin since it is connection-specific.